codemp/client/cl_uiapi.cpp

```diff
@@ -36,6 +39,62 @@ extern botlib_export_t *botlib_export;
 static uiExport_t *uie; // ui export table
 static vm_t *uivm; // ui vm, valid for legacy and new api
 
+typedef std::unordered_map<g2handle_t, CGhoul2Info_v*> g2HandleToG2_m;
+
+static g2HandleToG2_m g2Mapping;
+static g2handle_t g2NextHandle = (g2handle_t)1; // Start at 1, because 0 has special meaning
+
+static inline CGhoul2Info_v *CL_UI_G2Map_GetG2FromHandle( g2handleptr_t g2h )
```

Comment:
Instead of duplicating all this g2mapping code, it would be nice to wrap it in a class so it can be instantiated twice